### ISSUE TITLE
diff-cover takes entire project as diff

### ISSUE THREAD

**Issue by Tomasz-Kluczkowski** [Bachmann1234/diff-cover#91]
Hi,

I use diff-cover successfully and  with great pleasure for my personal projects and tried implementing it at work to ease development. 
Unfortunately despite checking that the diff is empty (branch is a fresh copy of master) when I run diff-cover on the previously generated coverage.xml (on the unchanged copy of master) it will report every single file in the coverage.xml that is missing coverage as missing, completely ignoring the fact that there is no diff and the report should be empty!

The projects structure is quite nested.
main_dir/sub_dir/src/project/code_folder_1 etc.

tests are in:
main_dir/sub_dir/tests

I am running diff-cover from main_dir.

Do you think there can be a problem with file paths ?

Btw. the same problem applies to diff-quality. Also it was unable to see .flake8 config file present in the folder from which I am running diff-quality command.

Many thanks for looking into this :).

Tom

**Comment by Bachmann1234** [comment]
Ok, so what is the specific command you are running (what args) and form what directory.

also, you say there is no diff but humor me and see if this has any output

`git -c diff.mnemonicprefix=no diff origin/master...HEAD --no-color  --no-ext-diff`

**Comment by Tomasz-Kluczkowski** [comment]
Hi Matt,

Many thanks for coming back so fast and pointing me at the right direction.

I have checked the command I am using and it is:
diff-cover coverage.xml —compare-branch=“$COMPARE_BRANCH” —html-report “$COV_PATH”

Where:
COMPARE_BRANCH=‘origin/develop’ - here was the mistake, it should not be a string :)!

I have fixed it and all is good!

Once again many thanks for your help and wonderful open source project!

 Best regards,

Tomasz

On 8 Feb 2019, at 02:22, Matt Bachmann <notifications@github.com<mailto:notifications@github.com>> wrote:


Ok, so what is the specific command you are running (what args) and form what directory.

also, you say there is no diff but humor me and see if this has any output

git -c diff.mnemonicprefix=no diff origin/master...HEAD --no-color --no-ext-diff
